src/write.rs

```diff
@@ -187,9 +197,76 @@ impl<W> WavWriter<W>
 
     /// Writes the RIFF WAVE header
     fn write_header(&mut self) -> io::Result<()> {
+        if self.extensible {
+            self.write_waveformatextensible()
+        } else {
+            self.write_waveformatex()
+        }
+    }
+
+    /// Writes the header based on WAVEFORMATEX struct
+    fn write_waveformatex(&mut self) -> io::Result<()> {
         // Useful links:
-        // https://msdn.microsoft.com/en-us/library/ms713462.aspx
         // https://msdn.microsoft.com/en-us/library/ms713497.aspx
+        // http://soundfile.sapp.org/doc/WaveFormat/
+        let mut header = [0u8; 44];
+        let spec = &self.spec;
+
+        // Write the header in-memory first.
+        {
+            let mut buffer: io::Cursor<&mut [u8]> = io::Cursor::new(&mut header);
```

Comment:
As `WAVEFORMATEXTENSIBLE` includes a `WAVEFORMATEX`, we could extract writing this part into a method, so it could be used to write both headers.